src/game-engine/positions/decode.test.ts

```diff
@@ -33,7 +34,9 @@ const testEncodeDecode = (pledge) => {
   it(`${pledge.constructor.name} is the same after encoding and decoding`, () => {
     const encoded = pledge.toHex();
     const decoded = decode(encoded);
-    expect(decoded).toEqual(pledge);
+    // We need to use JSON stringify due to the BN.js having possible different 
+    // internal representations of the same number
+    expect(JSON.stringify(decoded)).toEqual(JSON.stringify(pledge));
```

Comment:
This seems a bit worrying to me. Does this mean we can't reliably do `a === b` in the code?

Reply:
I think either way we want to avoid using `===` for objects as in Javascript that is checking if they're the same reference not have the same values (which is what we're interested in here). I think this issue will mostly come up in testing as the `expect.areEqual` does a pretty deep equality comparison that can fail on BN.js numbers.

It might aslo fail on https://lodash.com/docs/4.17.10#isEqual as well